kinesis-lambda-error-handling/README.md

```diff
@@ -0,0 +1,108 @@
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) Installed
+* [AWS SAM](https://aws.amazon.com/serverless/sam/) The AWS Serverless Application Model (SAM) is an open-source framework for building serverless applications.
+
+* [AWS Cloud9](https://aws.amazon.com/cloud9/) Other alternative is to use Cloud IDE which comes with prepackaged programming languages, CLIs needed for this project.
+
+## Deployment Instructions
+
+1. Clone the project to your local working directory
+
+   ```sh
+   git clone https://github.com/aws-samples/serverless-patterns/ 
+   ```
+
+2. Change the working directory to this pattern's directory
+
+   ```sh
+   cd serverless-patterns/kinesis-lambda-error-handling
+   ```
+3. From the command line, use AWS SAM to build and deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```
+   sam build
+   sam deploy --guided
+   ```
+4. During the prompts:
+
+   - Enter a stack name
+   - Enter the desired AWS Region
+   - Allow SAM CLI to create IAM roles with the required permissions.
+
+   Once you have run `sam deploy --guided` mode and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in the future to use these defaults.
+   Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+![Reference Architecture](/images/kinesis-lambda-error-handling.jpg)
+
+The pattern builds infrastructure with Amazon Kinesis data stream, a consumer Lambda function, SQS queue to load the failed records for further troubleshooting, and cloudwatch to validate the logs for the success, failure with retries.
+
+The event source mapping that reads records from your Kinesis stream, invokes AWS Lambda function synchronously, and retries on errors. If Lambda throttles the function or returns an error without invoking the function, Lambda retries until the records expire or exceed the maximum age that you configure on the event source mapping.
+
+## Testing 
+To test the pattern, we are breaking the test cases into two scenarios.
+
+### Scenario 1: Put messages without the Poision pill
+- test-records-without-poison-pill.json --> Holds the base64 data with partition key
+- without-poison-pill-put-records.sh --> bash script to put records into Kinesis data stream
+
+```
+chmod +x kinesis-producer/*.sh;
+./kinesis-producer/without-poison-pill-put-records.sh
+```
+
+- Navigate to AWS Console, and then to Cloudwatch, Log groups, select the log group and the latest Log stream
```

Comment:
what about using sam logs to do this in the CLI? As these instructions are not quite clear of what we should be looking at.